packages/core/src/odata/common/entity-serializer.ts

```diff
@@ -81,26 +95,56 @@ export function entitySerializer(tsToEdm) {
     }, {});
   }
 
-  function serializeComplexTypeField<EntityT extends EntityBase>(
-    complexTypeField: ComplexTypeField<EntityT>,
+  function serializeComplexTypeFieldLegacy<
```

Comment:
also here, a todo and deprecated keyword would be helpful

Reply:
check